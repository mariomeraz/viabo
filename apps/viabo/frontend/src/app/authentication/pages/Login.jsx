import { useEffect } from 'react'

import { Box, Stack, Typography } from '@mui/material'
import { useQueryClient } from '@tanstack/react-query'
import { LazyLoadImage } from 'react-lazy-load-image-component'

import { LoginForm } from '../components'

import LoginImage from '@/shared/assets/img/login-image-2x.webp'
import { Page } from '@/shared/components/containers'
import { FullLogo } from '@/shared/components/images'

const Login = () => {
  const client = useQueryClient()

  useEffect(() => {
    client.removeQueries()
  }, [])

  return (
    <Page title="Inicio de SesiÃ³n">
      <Stack alignItems={'center'} justifyContent={'center'} minHeight={'100dvH'}>
        <Stack px={{ sm: 10, xl: 20 }} width={1} height={1} direction={'row'}>
          <Stack
            flexGrow={1}
            width={1}
            height={1}
            minHeight={'70vh'}
            maxHeight={'90vH'}
            position={'relative'}
            sx={{ overflowX: 'hidden', overflowY: 'auto' }}
            justifyContent={'space-between'}
          >
            <Stack direction={'row'} spacing={1} alignItems={'center'} px={{ xs: 5, sm: 0 }}>
              <FullLogo sx={{ width: 100 }} />
              <Typography fontWeight={'600'} variant="subtitle1">
                Agilidad en pagos
              </Typography>
            </Stack>
            <Box flexGrow={1} />
            <Stack flex={1} px={{ xs: 0, sm: 5, xl: 15 }}>
              <LoginForm />
            </Stack>
          </Stack>
          <Stack
            width={1}
            justifyContent={'center'}
            height={1}
            flexGrow={1}
            sx={{ display: { xs: 'none', sm: 'none', lg: 'flex' } }}
          >
            <Box
              component={LazyLoadImage}
              placeholderSrc="https://zone-assets-api.vercel.app/assets/img_placeholder.svg"
              wrapperClassName="wrapper"
              effect={'blur'}
              height={1}
              maxHeight={'90vH'}
              sx={{ objectFit: 'fill' }}
              borderRadius={4}
              src={LoginImage}
            />
          </Stack>
        </Stack>
      </Stack>
    </Page>
  )
}

export default Login
