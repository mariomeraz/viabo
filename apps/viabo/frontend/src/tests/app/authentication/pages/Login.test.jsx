import { describe, expect, it } from 'vitest'

import Login from '@/app/authentication/pages/Login'
import { render } from '@/tests/testUtils'

describe('Pagina Inicio de Sesión', () => {
  it('Debería mostrar la pagina de inicio de sesión', async () => {
    const { container } = render(<Login />)
    expect(container).toMatchSnapshot()
  })
})
