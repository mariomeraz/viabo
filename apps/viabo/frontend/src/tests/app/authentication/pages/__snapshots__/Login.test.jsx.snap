// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagina Inicio de Sesión > Debería mostrar la pagina de inicio de sesión 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiBox-root css-10klw3m"
  >
    <div
      class="MuiStack-root css-tk14zv-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1ibyhwp-MuiStack-root"
      >
        <div
          class="MuiStack-root css-10548l2-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1cbb48g-MuiStack-root"
          >
            <div
              class="MuiBox-root css-1rfvmq9"
            >
              <svg
                data-name="Capa 2"
                fill="#B9FF00"
                id="Capa_2"
                viewBox="0 0 73.29 20.13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 1"
                  id="Layer_1"
                >
                  <path
                    d="M45.24 6.55c.18-.11.25-.14.32-.19 3.23-2.57 9.51-1.33 10.84 4.15.37 1.55.36 3.09-.17 4.59-.97 2.77-3.03 4.24-5.85 4.73-1.29.23-2.57.2-3.83-.2-2.98-.93-4.92-3.52-4.95-6.69-.03-4.04-.03-8.09-.04-12.13V.68c.02-.61.08-.67.69-.68h2.16c.81 0 .82 0 .82.8v5.75Zm7.61 5.83c-.03-.22-.06-.55-.13-.87-.39-1.8-2-3.11-3.76-3.06a3.83 3.83 0 0 0-3.66 3.18c-.17.97-.03 1.92.38 2.82.64 1.39 2.18 2.23 3.71 2.05 1.88-.22 3.39-1.56 3.46-4.12Zm-16.49 6.61c-.28.14-.52.24-.75.36-4.44 2.28-10.12-.71-10.81-5.7-.59-4.21 2.15-7.93 6.33-8.68 4.41-.78 8.53 2.65 8.76 6.92.09 1.7.05 3.41.07 5.11v1.97c0 .74-.05.79-.77.8-.5 0-1.01-.01-1.51 0-.6.02-1.14-.04-1.32-.79Zm-.06-6.52c-.04-.83-.23-1.63-.66-2.37-1.47-2.47-4.84-2.64-6.56-.33-1.35 1.82-1.1 4.56.57 6.1 1.07.98 3.21 1.59 4.88.25 1.17-.94 1.72-2.17 1.77-3.66Zm29.38 7.54c-4.2-.03-7.59-3.45-7.54-7.63.04-4.18 3.47-7.53 7.68-7.5 4.1.03 7.48 3.45 7.48 7.57 0 4.19-3.43 7.59-7.61 7.56Zm4-7.53c0-1.3-.44-2.41-1.36-3.32-1.52-1.51-3.83-1.5-5.36-.01-1.9 1.86-1.68 4.7-.18 6.42 1.08 1.24 3.53 1.93 5.26.41 1.06-.93 1.62-2.09 1.64-3.5ZM7.59 18.25c-.2-.32-.32-.5-.42-.68-.66-1.13-1.29-2.28-1.98-3.39-.27-.44-.23-.78 0-1.2.96-1.71 2.12-3.25 3.55-4.59.48-.45.97-.88 1.45-1.33.38-.35.37-.37.12-.83-.23-.41-.47-.82-.74-1.3h8.69c-1.44 2.5-2.85 4.92-4.27 7.38-.63-.42-.67-1.19-1.28-1.64-2.35 2.04-4.18 4.41-5.12 7.6Zm15.52-5.82v6.42c0 .82-.01.83-.86.83h-2.16c-.61 0-.68-.07-.68-.68V5.96c0-.75.04-.79.76-.8h2.23c.65 0 .71.06.71.72v6.55ZM0 4.9c.2-.01.38-.04.57-.04 1.33 0 2.66.01 3.99-.01.44 0 .73.11.93.53.48 1 1.02 1.98 1.5 2.98.07.14.07.42-.03.53-1 1.18-1.88 2.46-2.61 3.84C3.94 12.37.24 5.72-.01 4.9Z"
                  />
                  <path
                    d="M10.8 13.68c1.17 2.03 2.27 3.92 3.41 5.89-.25.03-.41.07-.57.07H9.58c-.76 0-1.16-.44-1.11-1.19.01-.17.04-.35.1-.51.43-1.36 1.07-2.62 1.9-3.78.09-.12.17-.25.34-.48Z"
                  />
                </g>
              </svg>
            </div>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-a2smzr-MuiTypography-root"
            >
              Agilidad en pagos
            </h6>
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          />
          <div
            class="MuiStack-root css-3elc6d-MuiStack-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiStack-root css-4lvoef-MuiStack-root"
              >
                <div
                  class="MuiStack-root css-nen11g-MuiStack-root"
                >
                  <div>
                    <h3
                      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-1a9rpoj-MuiTypography-root"
                    >
                      Aquí Comienza
                    </h3>
                  </div>
                  <div>
                    <h3
                      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-17i1ark-MuiTypography-root"
                    >
                      tu agilidad en pagos
                    </h3>
                  </div>
                </div>
                <div
                  style="min-height: 15px; opacity: 0;"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1au37kk-MuiTypography-root"
                  />
                </div>
                <div
                  style="opacity: 0;"
                >
                  <form
                    action="#"
                  >
                    <div
                      class="MuiStack-root css-ib3th8-MuiStack-root"
                    >
                      <div
                        class="MuiStack-root css-jfdv4h-MuiStack-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1cv7szy-MuiFormLabel-root-MuiInputLabel-root"
                          for="email"
                        >
                          Email
                        </label>
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-10k2l18-MuiFormControl-root-MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1622mpx-MuiInputBase-root-MuiOutlinedInput-root"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input css-fw5cm8-MuiInputBase-input-MuiOutlinedInput-input"
                              id="email"
                              name="email"
                              placeholder="usuario@dominio.com"
                              type="email"
                              value=""
                            />
                            <fieldset
                              aria-hidden="true"
                              class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="css-ihdtdm"
                              >
                                <span
                                  class="notranslate"
                                >
                                  ​
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiStack-root css-jfdv4h-MuiStack-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1cv7szy-MuiFormLabel-root-MuiInputLabel-root"
                          for="password"
                        >
                          Contraseña
                        </label>
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-56wmv2-MuiInputBase-root-MuiOutlinedInput-root"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-qhnzi4-MuiInputBase-input-MuiOutlinedInput-input"
                              id="password"
                              name="password"
                              placeholder="********"
                              type="password"
                              value=""
                            />
                            <div
                              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-21twve-MuiInputAdornment-root"
                            >
                              <button
                                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1w1roue-MuiButtonBase-root-MuiIconButton-root"
                                tabindex="0"
                                type="button"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  stroke="currentColor"
                                  stroke-width="0"
                                  viewBox="0 0 256 256"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M234.42,162A12,12,0,1,1,213.58,174l-16.86-29.5a127.19,127.19,0,0,1-30.17,13.86L171.84,190a12,12,0,0,1-9.87,13.8,11.22,11.22,0,0,1-2,.17,12,12,0,0,1-11.82-10L143,163.17a136.5,136.5,0,0,1-30.06,0L107.84,194A12,12,0,0,1,96,204a11.22,11.22,0,0,1-2-.17A12,12,0,0,1,84.16,190l5.29-31.72a127.19,127.19,0,0,1-30.17-13.86L42.42,174A12,12,0,1,1,21.58,162L40,129.85a159.73,159.73,0,0,1-17.31-18.31A12,12,0,0,1,41.34,96.46C57.38,116.32,85.44,140,128,140s70.62-23.68,86.66-43.54a12,12,0,0,1,18.67,15.08A159.73,159.73,0,0,1,216,129.85Z"
                                  />
                                </svg>
                                <span
                                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                />
                              </button>
                            </div>
                            <fieldset
                              aria-hidden="true"
                              class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="css-ihdtdm"
                              >
                                <span
                                  class="notranslate"
                                >
                                  ​
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiStack-root css-1v3364e-MuiStack-root"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-1c0xctp-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                          id=":r3:"
                          tabindex="0"
                          type="submit"
                        >
                          Accesar a mi cuenta
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <p
                        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-shwflf-MuiTypography-root"
                      >
                        ¿No tienes una cuenta?  
                        <a
                          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1xk92bt-MuiTypography-root-MuiLink-root"
                          href="/registro"
                        >
                          Inscribete.
                        </a>
                      </p>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-glg4a-MuiStack-root"
        >
          <span
            class="wrapper lazy-load-image-background blur"
            style="background-image: url(https://zone-assets-api.vercel.app/assets/img_placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
          >
            <img
              class="MuiBox-root css-1buctor"
              src="/src/shared/assets/img/login-image-2x.webp"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
